runelite-api/src/main/java/net/runelite/api/widgets/Widget.java

```diff
@@ -452,4 +457,42 @@
 	 * @return the actions
 	 */
 	String[] getActions();
+
+	/**
+	 * Creates a dynamic widget child
+	 *
+	 * @param index the index of the new widget in the children list or -1 to append to the back
+	 * @param type  the type of the widget
+	 */
+	Widget createChild(int index, int type);
+
+	/**
+	 * Creates a menu action on the widget
+	 *
+	 * @param index The index of the menu
+	 * @param action The string to be displayed next to the widget's name in the context menu
+	 */
+	void setAction(int index, String action);
+
+	/**
+	 * Sets a script to be ran when the a menu action is clicked.
+	 * hasListener must be true for this to take effect
+	 *
+	 * @param args A ScriptID, then the args for the script
+	 */
+	void setOnOpListener(Object ...args);
+
+	/**
+	 * If this widget has any listeners on it
+	 */
+	boolean hasListener();
+
+	/**
+	 * Sets if the widget has any listeners. This should be called whenever a setXListener function is called
+	 */
+	void setHasListener(boolean hasListener);
+
+	boolean hasScript();
```

Comment:
Document this too

Reply:
I honestly have no idea what this does, hence why it is undocumented.

Reply:
Then at least document that, maybe even add TODO